kubernetes/client.go

```diff
@@ -79,6 +79,10 @@ type IstioClient struct {
 	k8s                *kube.Clientset
 	istioConfigApi     *rest.RESTClient
 	istioNetworkingApi *rest.RESTClient
+	// isOpenShift can be cached per client
```

Comment:
Again, I would like to see a comment here to explicitly explain to the user why this is a pointer - we need to explain to devs looking at this code that it is expected and a very valid value that this variable be nil (when isOpenShift has yet to be determined).